src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -137,74 +146,120 @@ public String getScript() {
-                interpreter,
-                scriptPath,
-                logFile,
-                resultFile, resultFile, resultFile);
-        }
-        cmd = cmd.replace("$", "$$"); // escape against EnvVars jobEnv in LocalLauncher.launch
+        envVars.put(cookieVariable, "please-do-not-kill-me");
 
-        if (LAUNCH_DIAGNOSTICS) {
-            args.addAll(Arrays.asList("sh", "-c", cmd));
+        String arch = ws.act(new getArchitecture());
+        List<String> launcherCmd = null;
+        String launcherBinary = LAUNCHER_PREFIX + os.toString() + arch;
+        InputStream launcherStream = DurableTask.class.getResourceAsStream(launcherBinary);
+        if ((launcherStream != null) && !FORCE_SHELL_WRAPPER) {
+            FilePath controlDir = c.controlDir(ws);
+            FilePath launcherAgent = controlDir.child(launcherBinary);
+            launcherAgent.copyFrom(launcherStream);
+            launcherAgent.chmod(0755);
+            launcherCmd = binaryLauncherCmd(c, ws, shell,
+                                            controlDir.getRemote(),
+                                            launcherAgent.getRemote(),
+                                            scriptPath,
+                                            cookieValue,
+                                            cookieVariable);
         } else {
-            // JENKINS-58290: launch in the background. Also close stdout/err so docker-exec and the like do not wait.
-            args.addAll(Arrays.asList("sh", "-c", "(" + cmd + ") >&- 2>&- &"));
+            String scriptString = scriptLauncherCmd(c, ws, interpreter, scriptPath, cookieValue, cookieVariable);
+            scriptString = scriptString.replace("$", "$$"); // escape against EnvVars jobEnv in LocalLauncher.launch
+            launcherCmd = new ArrayList<>();
+            if (LAUNCH_DIAGNOSTICS) {
+                launcherCmd.addAll(Arrays.asList("sh", "-c", scriptString));
+            } else {
+                // JENKINS-58290: launch in the background. Also close stdout/err so docker-exec and the like do not wait.
+                launcherCmd.addAll(Arrays.asList("sh", "-c", "(" + scriptString + ") >&- 2>&- &"));
+            }
         }
-        LOGGER.log(Level.FINE, "launching {0}", args);
-        Launcher.ProcStarter ps = launcher.launch().cmds(args).envs(escape(envVars)).pwd(ws).quiet(true);
+
+        LOGGER.log(Level.FINE, "launching {0}", launcherCmd);
+        Launcher.ProcStarter ps = launcher.launch().cmds(launcherCmd).envs(escape(envVars)).pwd(ws).quiet(true);
         if (LAUNCH_DIAGNOSTICS) {
             ps.stdout(listener);
         } else {
             ps.readStdout().readStderr(); // TODO RemoteLauncher.launch fails to check ps.stdout == NULL_OUTPUT_STREAM, so it creates a useless thread even if you never called stdout(…)
+            LOGGER.log(Level.WARNING,"Durable Task launching wrapper process in the background. An init process must be used to reap the orphaned wrapper process.");
```

Comment:
IIUC, this will log a warning to the Jenkins system log for every shell step in every Pipeline, even if there is no problem. Since that will blow up the logs and there's nothing the admin can do to "fix" the warnings, I think it would make more sense to document this in the `sh` step over in workflow-durable-task and remove this warning entirely.

Reply:
Done in ef70f1891.